src/MultiCollateralVault.sol

```diff
@@ -0,0 +1,275 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.13;
+
+import "yield-utils/math/WDiv.sol";
+import "yield-utils/math/WMul.sol";
+import "yield-utils/token/IERC20.sol";
+import "yield-utils/token/IERC20Metadata.sol";
+import "yield-utils/token/TransferHelper.sol";
+import "yield-utils/access/Ownable.sol";
+import "yield-utils/access/AccessControl.sol";
+
+import "chainlink/v0.8/interfaces/AggregatorV3Interface.sol";
+
+library RMul {
+    function rmul(uint256 x, uint256 y) internal pure returns(uint256 z) {
+        z = x * y;
+        unchecked {z /= 1e27;}
+    }
+}
+
+library RDiv {
+    function rdiv(uint256 x, uint256 y) internal pure returns(uint256 z) {
+        z = (x*1e27) / y;
+    }
+}
+
+
+/// https://github.com/yieldprotocol/mentorship2022/issues/6
+
+/// @title MultiCollateralVault
+/// @author hashedMae
+/// @notice A contract for borrowing WETH against deposits of tokens that have been added as approved collateral
+/// @dev Additional collaterals can be added after deployment but only WETH can ever be borrowed
+
+contract MultiCollateralVault is AccessControl {
+    
+    using TransferHelper for IERC20;
+
+    event Deposit(address indexed user, IERC20 indexed token, uint256 amount);
+    event Borrow(address indexed user, uint256 amount);
+    event Repay(address indexed user, uint256 repaid, uint256 remaining);
+    event Withdraw(address indexed user, IERC20 indexed token, uint256 amount);
+
+    /// mainnet 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2
+    /// rinkeby 0xDf032Bc4B9dC2782Bb09352007D4C57B75160B15
+    IERC20 WETH;
+
+    /// @dev nested mapping for tracking user WETH debt
+    mapping(address =>  uint256) public  debt;
+
+    /// @dev nested mapping for tracking user deposits, first address is user address, second is token address
+    /// mapping(user => mapping(token => deposit))
+    mapping(address => mapping(IERC20 => uint256)) public  deposits;
+
+    mapping(IERC20 => CollateralInfo) public collateralInfos;
+
+    struct CollateralInfo {
+        AggregatorV3Interface oracle;
+        uint96 ratio;
+    }
+
+   /// CollateralInfo collateralInfo;
+
+    IERC20[] public Collaterals;
+
+    ///@param WETH_ address for the WETH contract
+    constructor (IERC20 WETH_) {
+        WETH = WETH_;
+    }
+
+/***
+-------------------------------------------------------------------------------------------------------------------------------------------------------------
+    Admin Functions
+-------------------------------------------------------------------------------------------------------------------------------------------------------------
+ */
+
+    /// @dev checks to see if a token is currently accepted as collteral and if it has been added but not yet live cause there is no ratio set yet
+    function addCollateral(IERC20 token_, AggregatorV3Interface oracle_) external auth {
```

Comment:
It seems to me that it would be simpler to set the oracle and collateralization ratio in the same function, since one doesn't make sense without the other.